### PR TITLE
[CAP-282] Get all application for a company

### PR DESCRIPTION
**Pull request by BuiMinhKhoi**

### CODE REVIEW

File: src/main/java/org/capstone/job_fair/repositories/attendant/cv/ApplicationRepository.java [hunk 1]
```diff
@@ -5,11 +5,29 @@
...
+
 @Repository
 public interface ApplicationRepository extends JpaRepository<ApplicationEntity, String> {
 
     Page<ApplicationEntity> findAllByCreateDateBetweenAndStatusAndCvAttendantAccountId(long fromDate, long toDate, Application status, String id, Pageable pageable);
 
+    @Query("select a from ApplicationEntity a " +
+            "where a.registrationJobPosition.companyRegistration.companyId = ?1 " +
+            "and a.registrationJobPosition.id = ?2 and a.status in ?3")
+    Page<ApplicationEntity> findAllApplicationOfCompanyByJobPositionIdAndStatusIn(String companyId, String jobPositionId, List<Application> applicationList, Pageable pageable);
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/repositories/attendant/cv/ApplicationRepository.java:21]: Refactor status please. Use name for parameter please

File: src/main/java/org/capstone/job_fair/services/impl/attendant/cv/ApplicationServiceImpl.java [hunk 2]
```diff
@@ -45,6 +47,25 @@ private boolean isJobPositionExist(String registrationId) {
         return regisJobPosRepository.countById(registrationId) != 0;
     }
 
+    private void validatePaging(int pageSize, int offset) {
+        if (offset < DataConstraint.Paging.OFFSET_MIN || pageSize < DataConstraint.Paging.PAGE_SIZE_MIN)
+            throw new IllegalArgumentException(MessageUtil.getMessage(MessageConstant.Application.INVALID_PAGE_NUMBER));
+    }
+
+
+    private String ApplicationForCompanySortByMapper(String sortBy) {
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/services/impl/attendant/cv/ApplicationServiceImpl.java:56]: Fix name

File: src/main/java/org/capstone/job_fair/services/impl/attendant/cv/ApplicationServiceImpl.java [hunk 3]
```diff
@@ -64,17 +85,44 @@ public ApplicationDTO createNewApplication(ApplicationDTO dto) {
...
+    public Page<ApplicationForCompanyResponse> getApplicationOfCompanyByJobFairIdAndStatus(String companyId, String jobFairId, List<Application> statusList, int pageSize, int offset, String sortBy, Sort.Direction direction) {
+        validatePaging(pageSize, offset);
+        sortBy = ApplicationForCompanySortByMapper(sortBy);
+        if (statusList == null || statusList.isEmpty()) {
+            statusList = Arrays.asList(Application.FINISHED, Application.PENDING, Application.CANCEL, Application.APPROVE, Application.REJECT);
+        }
+
+        Page<ApplicationEntity> applicationEntityPage = applicationRepository.findAllApplicationOfCompanyByJobFairIdAndStatusIn
+                (companyId, jobFairId, statusList, PageRequest.of(offset, pageSize).withSort(direction, sortBy));
+        Page<ApplicationForCompanyResponse> responses = applicationEntityPage.map(entity -> applicationMapper.toApplicationForCompanyResponse(entity));
```
Comment (TienTruong2000) [inline src/main/java/org/capstone/job_fair/services/impl/attendant/cv/ApplicationServiceImpl.java:109]: Mapping response should move to controller

**Review by TienTruong2000 — CHANGES_REQUESTED** [review]

**Comment by BuiMinhKhoi** [comment]
Screen Shot 2022-03-20 at 5 20 52 PM

**Review by TienTruong2000 — APPROVED** [review]
LGTM

**Pull request merged**